### ISSUE TITLE
Run integration tests for Zalenium-Kubernetes 

### ISSUE THREAD

**Issue by diemol** [zalando/zalenium#157]
The execution should work in a similar way as when we execute them in docker. This means that it should work locally and in Travis, using Minikube is enough.

**Comment by pearj** [comment]
Ooh didn't realise that travis allows you to run integration tests. 

So I guess we install minikube into the travis vm?

The fabric8 maven plugin knows how to deploy to kubernetes. Could be worth using. https://maven.fabric8.io

**Comment by pearj** [comment]
Ok so running minikube inside the travis vm is not going to work, because travis doesn't let you run [nested VMs](https://github.com/travis-ci/travis-ci/issues/6060).

So basically you'd have to figure out how to run kubernetes without having a VM to do it, and from what I'm told kubernetes is super fiddly to get running because you have to generate certificates and such things.

OpenShift can run as a privileged docker container, which is one option.  And it looks like kubernetes has another install option with "[conjure-up](https://kubernetes.io/docs/getting-started-guides/ubuntu/local/)".  I might see what happens in an Ubuntu 14.04 VM (the newest that travis seems to allow) with kubernetes conjure up.

**Comment by pearj** [comment]
Ok, conjure-up is not going to work in travis, because it needs snapd, which installs a 4.4 linux kernel, which isn't [allowed in travis](https://forum.snapcraft.io/t/support-snaps-in-travis-trusty/421).  So, we're back to square 1.

OpenShift might be the easiest option, presuming that it works.

**Comment by pearj** [comment]
Hmm, [running a container](https://docs.openshift.org/latest/getting_started/administrators.html#running-in-a-docker-container) didn't work.  It does say that `This method is supported on Fedora, CentOS, and Red Hat Enterprise Linux (RHEL) hosts only.` So it's probably no great surprise that it fails with Ubuntu 14.04.

It got close, but then fell over with:
```
...
I0619 04:35:18.613786    5089 start_master.go:839] Started "openshift.io/deployer"
I0619 04:35:18.613958    5089 factory.go:84] Starting deployer controller
E0619 04:35:18.661184    5089 docker_service.go:190] Failed to execute Info() call to the Docker client: Error reading remote info: json: cannot unmarshal number into Go value of type bool
W0619 04:35:18.661266    5089 docker_service.go:191] Falling back to use the default driver: "cgroupfs"
F0619 04:35:18.661402    5089 node.go:297] failed to run Kubelet: failed to create kubelet: misconfiguration: kubelet cgroup driver: "systemd" is different from docker cgroup driver: "cgroupfs"

```

It looks like kuberentes integration tests for zalenium will take a bit of effort then...

**Comment by diemol** [comment]
> Ooh didn't realise that travis allows you to run integration tests.

Well, it is just to start the docker image and run a few Selenium tests. I hope to do something similar with Kubernetes.

We need to research a bit, at the end we must have something where we can rely, because just testing locally isn't enough. It is quite nice when you can test everything from in a fresh VM.

I'll invest some time in that later during the week.

**Comment by pearj** [comment]
Ooh, apparently snaps work in travis somehow now: https://forum.snapcraft.io/t/support-snaps-in-travis-trusty/421/6

**Comment by diemol** [comment]
Oh, I was checking this example https://github.com/aaron-prindle/minikube-travis-example

**Comment by pearj** [comment]
even better

**Comment by diemol** [comment]
Implemented via #257
